storage3/_async/bucket.py

```diff
@@ -25,26 +23,21 @@ async def _request(
         url: str,
         json: Optional[dict[Any, Any]] = None,
     ) -> Response:
-        response = await self._client.request(
-            method, url, headers=self.headers, json=json
-        )
+        response = await self._client.request(method, url, json=json)
         try:
             response.raise_for_status()
         except HTTPError:
-            raise StorageException(response.json())
+            raise StorageException(
+                {**response.json(), "statusCode": response.status_code}
+            )
 
         return response
 
     async def list_buckets(self) -> list[AsyncBucket]:
         """Retrieves the details of all storage buckets within an existing product."""
         # if the request doesn't error, it is assured to return a list
-        res = await self._request("GET", f"{self.url}/bucket")
-        return [
-            AsyncBucket(
-                **bucket, _url=self.url, _headers=self.headers, _client=self._client
-            )
-            for bucket in res.json()
-        ]
+        res = await self._request("GET", f"{self._client.base_url}bucket")
```

Comment:
```suggestion
        res = await self._request("GET", f"/bucket")
```
`base_url` need not be added here explicitly as it is set on the httpx client itself when you did `AsyncClient(base_url=base_url)`; now all requests should be relative to that base URL